src/lib/y2partitioner/actions/move_partition.rb

```diff
@@ -0,0 +1,344 @@
+      # Checks whether it is possible to move the partition, and if so, the action is performed.
+      #
+      # @note An error popup is shown when the partition cannot be moved. In case the partition
+      #   can be moved, a dialog is presented to select whether to move forward or backward.
+      #
+      # @return [Symbol] :finish if the action is performed; :back or dialog result otherwise.
+      def run
+        return :back unless validate
+
+        answer = move_partition_question
+        return answer unless answer == :ok
+
+        move_partition
+        :finish
+      end
+
+    private
+
+      # @return [Y2Storage::BlkDevice]
+      attr_reader :partition
+
+      # @return [Symbol] :forward, :backward
+      attr_reader :movement
+
+      # Checks whether the partition can be moved
+      #
+      # @note When the partition cannot be moved, it shows a popup with the reason,
+      #   see {#error}.
+      #
+      # @return [Boolean] true if the partition can be moved; false otherwise.
+      def validate
+        return true unless error
+
+        Yast2::Popup.show(error, headline: :error)
+        false
+      end
+
+      # Why the partition cannot be moved
+      #
+      # @note A partition can be moved if is not an extended partition and it does
+      #   not exist on disk and there is adjancent free space where to move it. In
+      #   some cases, the move action could be called over a device that is not a
+      #   partition (i.e., when the selected element on a table is not a partition).
+      #   For that reason, the type of the device must be checked.
+      #
+      # @return [String, nil] nil when there is no error and the partition can be moved.
+      def error
+        device_type_error ||
+          extended_partition_error ||
+          existing_partition_error ||
+          no_space_error
+      end
+
+      # Error when a no partition device is tried to be moved
+      #
+      # @return [String, nil] nil the device is a partition
+      def device_type_error
+        return nil if partition.is?(:partition)
+
+        _("Hard disks, BIOS RAIDs and multipath\n" \
+          "devices cannot be moved.")
+      end
+
+      # Error when a extended partition is tried to be moved
+      #
+      # @return [String, nil] nil when the partition is not extended
+      def extended_partition_error
+        return nil unless partition.type.is?(:extended)
+
+        _("An extended partition cannot be moved.")
+      end
+
+      # Error when an existing partition is tried to be moved
+      #
+      # @return [String, nil] nil if the partition does not exist on disk
+      def existing_partition_error
+        return nil unless partition.exists_in_devicegraph?(DeviceGraphs.instance.system)
+
+        format(
+          # TRANSLATORS: error message where %{name} is replaced by a partition name (e.g. /dev/sda1)
+          _("The partition %{name} is already created on disk\n" \
+            "and cannot be moved."),
```

Comment:
maybe mention that To move it, it have to be deleted and created again? So user knows what to do.

Reply:
Again, the text was simply copied. I also would like to improve that messages, but not sure if we want to do it now.

Reply:
me neither, so please try to clarify it. As I think that it makes sense to give user some hint how to solve it.

Reply:
From the beginning, the first goal has been to mimic the old partitioner, so I would go with the messages just as the old partitioner. Anyway, I have added FIXME notes as you suggested.